everviz/plugins/summary_plot/summary_callback.py

```diff
@@ -73,32 +97,35 @@ def _get_data_lines(data, summary_key_list, line_list, x_filter, x_key, mode="ma
     :param x_filter: One of date|batch
     :param x_key: Opposite of x_filter
     """
+    # pylint: disable=too-many-locals
     colors = cycle(DEFAULT_PLOTLY_COLORS)
+    realizations = data["realization"].unique()
     traces = []
     for color, key in zip(colors, summary_key_list):
         # Select all data belonging to the current key.
-        key_data = data[key]
-
+        key_data = data[[key, "realization"]]
         for line in line_list:
             # Select all rows belonging the current batch or date.
             line_data = key_data.xs(line, level=x_filter, drop_level=True).reset_index()
-
             # Set the name of the plotted line.
             name = key
             if len(line_list) > 1:
                 name += f", {x_filter}:{line}"
-
-            traces.append(
-                go.Scatter(
-                    y=line_data[key],
-                    x=line_data[x_key],
-                    mode=mode,
-                    marker={"size": 10},
-                    name=name,
-                    showlegend=True,
+            # Plot each realization separately, adapting the hovertext.
+            for real in realizations:
+                line = line_data[line_data["realization"].isin([real])]
+                traces.append(
+                    go.Scatter(
+                        y=line[key],
+                        x=line[x_key],
+                        mode=mode,
+                        marker={"size": 10},
+                        name=name,
+                        showlegend=True,
+                        line={"color": color},
+                        hovertext=f"realization:{real}",
```

Comment:
Is this really needed if the legend contains the realization number? If yes then it should also contain the batch number and the key.